lib/Tests.js

```diff
@@ -255,35 +209,42 @@ TestsModule.prototype.getRecentComponentTests = async function(componentUuid)
 }
 
 
-TestsModule.prototype.listUserDrafts = async function(user_id)
-  {
-    var query = {"state": "draft"};
-    if(user_id) query["insertion.user.user_id"] = user_id;
-    var p = await db.collection(this._collection).aggregate(
-                            [ { '$match' : query },
-                              { '$sort' : { 'insertion.insertDate': -1 } },
-                              { '$lookup': {
-                                            from: this._formCollection,
-                                            localField: "formObjectId",
-                                            foreignField: "_id",
-                                            as: "form"   
-                                           }},
-                              // { '$project' : {formId: 1, formName: '$form[0].formName', insertDate: 'insertion.insertDate', user:'insertion.user'}}
-                            ])
-            .toArray();
+// for the Autocomplete Route
+TestsModule.prototype.findTestIdStartsWith = async function(id_string,formId,max)
+{
+  // binary version.
+  max = max || 10;
 
+  // sanitize string of all extra characters.
+  var q = id_string.replace(/[_-]/g,'');
 
-      // logger.info("listUserDrafts",user_id||"all",":");
-      // logger.info(p);
-      return p;
-  }
+  // pad with zeroes and "F"s.
+  var qlow = q.padEnd(24,'0');
+  var qhigh = q.padEnd(24,'F');
+  // logger.info(qlow,qhigh);
+  var bitlow = ObjectID(qlow);
+  var bithigh =  ObjectID(qhigh);
+  
+  var query = {_id:{$gte: bitlow, $lte: bithigh}};
+  if(formId) query.formId = formId;
 
-TestsModule.prototype.deleteDraft = async function(record_id)
-  {
-    // caller should check authorization
-    return await db.collection(this._collection).deleteOne({_id: ObjectID(record_id), state:'draft'});
+  // logger.info(qlow,qhigh,bitlow,bithigh);
+  logger.info(query,"QfindTestIdStartsWith");
+ 
+  var matches = await db.collection("tests")
+    .find(query)  // binary ObjectIds only
+    .project({"_id":1, 'componentUuid':1, 'formId':1})
+    .limit(max)
+    .toArray();
+  // if(matches.length>=max) return new Error("Too many entries"); // too many!
+  for(var m of matches) {
+    try {  m.componentUuid = MUUID.from(m.componentUuid).toString(); } catch {};
```

Comment:
If we expect errors to occur here, why do we let them go silently?  We should either handle the errors or document why we don’t handle them.

Reply:
I'll address this when I clean up the 'autocomplete' route which uses this function.  But yes, you're absolutely correct of course - shouldn't ever have silent errors.